lib/attrsets.nix

```
@@ -1999,4 +2017,8 @@ rec {
   # DEPRECATED
   zip = warn
     "lib.zip is a deprecated alias of lib.zipAttrsWith." zipAttrsWith;
+
+  # DEPRECATED
+  cartesianProductOfSets = warn
```

```suggestion
  cartesianProductOfSets = warnIf (isInOldestRelease 2405)
```

See https://github.com/NixOS/nixpkgs/issues/291939 why that's necessary